File: baekjoon/game/9656/main.cpp
#include <iostream>
#include <cstdio>
#include <cstring>

using namespace std;

int N;
int cache[1001];
int play(int n);
int main()
{
    memset(cache, -1, sizeof(cache));
    scanf("%d", &N);
    if(play(N) == 0) {
        printf("CY\n");
    } else {
        printf("SK\n");
    }
    return 0;
}

int play(int n)
{
    int& ret = cache[n];
    if(ret != -1) return ret;
    ret = 0;
    
    if(n-3>0 && !play(n-3)) ret = 1;
    else if(n-1>0 && !play(n-1)) ret = 1;
    return ret;
}

File: baekjoon/game/9656/Main.java
import java.util.*;

class Main {
    static int N;
    static int cache[];
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        cache = new int[1001];
        Arrays.fill(cache, -1);
        
        N = sc.nextInt();
        String ret = (play(N)==0?"CY":"SK");
        System.out.println(ret);
    }
    
    public static int play(int n) {
        
        int ret = cache[n];
        if(ret != -1) return ret;
        ret = 0;
        
        if(n-3>0 && play(n-3) == 0) ret = 1;
        if(n-1>0 && play(n-1) == 0) ret = 1;
        
        return cache[n] = ret;
    }
}